addons/payment/models/payment_acquirer.py

```diff
@@ -221,6 +221,31 @@ def _check_fee_var_within_boundaries(self):
             if any(not 0 <= fee < 100 for fee in (acquirer.fees_dom_var, acquirer.fees_int_var)):
                 raise ValidationError(_("Variable fees must always be positive and below 100%."))
 
+    #=== ONCHANGE METHODS ===#
+
+    @api.onchange('state', 'allow_tokenization')
+    def _onchange_state_allow_tokenization(self):
+        tokens = self.env['payment.token'].search([('acquirer_id', 'in', self.ids)])
+        if not tokens:
+            return
+
+        tokens_sub = [token.get_linked_records_info() for token in tokens]
+        if tokens_sub:
+            message = _(
+                "Please note that %s \"Payment Tokens\" linked to this acquirer will be archived. "
+                "This means that %s active subscriptions will need new payment methods.",
+                len(tokens), len(tokens_sub)
+            )
```

Comment:
This is not really modular.  IMHO, we shouldn't have any 'subscription' logic anywhere in this module :thinking:.
Furthermore, you don't know whether the linked records info might be something else than subscriptions, ...
Should be done either with a `tokens._get_archive_warning_message()` (or sthg like that) or an override of the onchange in subscription, but this would be less clear anyway.